packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -314,9 +314,9 @@ class _ExpansionPanelListState extends State<ExpansionPanelList> {
   }
 
   void _handlePressed(bool isExpanded, int index) {
-    widget.expansionCallback?.call(index, isExpanded);
 
     if (widget._allowOnlyOnePanelOpen) {
+
       final ExpansionPanelRadio pressedChild = widget.children[index] as ExpansionPanelRadio;
 
       // If another ExpansionPanelRadio was already open, apply its
```

Comment:
If I am not mistaken, this code closes already open panels when a new child is selected. 
What is the behavior when you open an already open panel? Would it actually close the current panel and re-open it?